src/main/java/com/code_intelligence/jazzer/mutation/mutator/proto/BuilderMutatorFactory.java

```diff
@@ -243,22 +230,81 @@ public Builder detach(Builder builder) {
+  private final HashMap<CacheKey, SerializingMutator<? extends Builder>> internedMutators =
       new HashMap<>();
 
+  private SerializingMutator<Any.Builder> createMutatorForAny(
+      AnySource anySource, MutatorFactory factory) {
+    HashMap<String, Integer> typeUrlToIndex = new HashMap<>(anySource.value().length);
+    for (int i = 0; i < anySource.value().length; i++) {
+      Message defaultInstance = TypeLibrary.getDefaultInstance(anySource.value()[i]);
+      typeUrlToIndex.put(getTypeUrl(defaultInstance), i);
+    }
+
+    return assemble(mutator
+        -> internedMutators.put(new CacheKey(Any.Builder.class, anySource), mutator),
+        Any.getDefaultInstance()::toBuilder, makeBuilderSerializer(Any.getDefaultInstance()),
+        ()
+            -> mutateSumInPlace(
+                // Corpus entries may contain Anys with arbitrary (and even invalid) messages, so we
+                // fall back to mutating the first message type if the type isn't recognized.
+                (Any.Builder builder)
+                    -> typeUrlToIndex.getOrDefault(builder.getTypeUrl(), 0),
+                stream(anySource.value())
+                    .map(messageClass -> {
+                      SerializingMutator<Message> messageMutator =
+                          (SerializingMutator<Message>) factory.createOrThrow(notNull(
+                              withExtraAnnotations(asAnnotatedType(messageClass), anySource)));
+                      return mutateProperty(
+                          (Any.Builder anyBuilder)
+                              -> {
+                            try {
+                              return anyBuilder.build().unpack(messageClass);
+                            } catch (InvalidProtocolBufferException e) {
+                              // This can only happen if the corpus contains an invalid Any.
+                              return TypeLibrary.getDefaultInstance(messageClass);
+                            }
+                          },
+                          messageMutator,
+                          (Any.Builder any, Message message) -> {
+                            any.setTypeUrl(getTypeUrl(message));
+                            any.setValue(message.toByteString());
+                          });
+                    })
+                    .toArray(InPlaceMutator[] ::new)));
+  }
+
+  private static String getTypeUrl(Message message) {
+    return "type.googleapis.com/" + message.getDescriptorForType().getFullName();
+  }
+
   @Override
   public Optional<SerializingMutator<?>> tryCreate(AnnotatedType type, MutatorFactory factory) {
     return asSubclassOrEmpty(type, Builder.class).map(builderClass -> {
-      if (internedMutators.containsKey(builderClass)) {
-        return internedMutators.get(builderClass);
+      AnySource anySource = type.getDeclaredAnnotation(AnySource.class);
+      Preconditions.require(anySource == null || anySource.value().length > 0,
```

Comment:
Also check that all given classes extend `Message`?

Reply:
This should be handled by https://github.com/CodeIntelligenceTesting/jazzer/pull/691/files#diff-8a6a842756beb4d84e4e92461e8095eacc5542cc31cb41e4f0129c07624a51c0R39.

Reply:
Ah, yes!